### ISSUE TITLE
Failed to build in Mac OS

### ISSUE THREAD

**Issue by lilidan** [buildbuddy-io/buildbuddy#435]
Run 
`bazelisk run -c opt server:buildbuddy --verbose_failures  `  command faild in Mac OS.

Log:

```
ERROR: /private/var/tmp/_bazel_bytedance/dec3efcbf31172670b2193c1360342f6/external/com_github_mattn_go_sqlite3/BUILD.bazel:3:11: GoCompilePkg external/com_github_mattn_go_sqlite3/go-sqlite3.a failed: (Exit 1): builder failed: error executing command 
  (cd /private/var/tmp/_bazel_bytedance/dec3efcbf31172670b2193c1360342f6/sandbox/darwin-sandbox/901/execroot/buildbuddy && \
  exec env - \
    APPLE_SDK_PLATFORM=MacOSX \
    APPLE_SDK_VERSION_OVERRIDE=11.1 \
    CC=external/local_config_cc/wrapped_clang \
    CGO_ENABLED=1 \
    GOARCH=amd64 \
    GOOS=darwin \
    GOPATH='' \
    GOROOT=external/go_sdk_darwin \
    GOROOT_FINAL=GOROOT \
    PATH=external/local_config_cc:/bin:/usr/bin \
    XCODE_VERSION_OVERRIDE=12.4.0.12D4e \
  bazel-out/host/bin/external/go_sdk_darwin/builder compilepkg -sdk external/go_sdk_darwin -installsuffix darwin_amd64 -src external/com_github_mattn_go_sqlite3/backup.go -src external/com_github_mattn_go_sqlite3/callback.go -src external/com_github_mattn_go_sqlite3/convert.go -src external/com_github_mattn_go_sqlite3/doc.go -src external/com_github_mattn_go_sqlite3/error.go -src external/com_github_mattn_go_sqlite3/sqlite3.go -src external/com_github_mattn_go_sqlite3/sqlite3_context.go -src external/com_github_mattn_go_sqlite3/sqlite3_func_crypt.go -src external/com_github_mattn_go_sqlite3/sqlite3_go18.go -src external/com_github_mattn_go_sqlite3/sqlite3_load_extension.go -src external/com_github_mattn_go_sqlite3/sqlite3_opt_preupdate.go -src external/com_github_mattn_go_sqlite3/sqlite3_opt_preupdate_omit.go -src external/com_github_mattn_go_sqlite3/sqlite3_opt_userauth_omit.go -src external/com_github_mattn_go_sqlite3/sqlite3_other.go -src external/com_github_mattn_go_sqlite3/sqlite3_solaris.go -src external/com_github_mattn_go_sqlite3/sqlite3_type.go -src external/com_github_mattn_go_sqlite3/sqlite3_usleep_windows.go -src external/com_github_mattn_go_sqlite3/sqlite3_windows.go -src external/com_github_mattn_go_sqlite3/static_mock.go -src external/com_github_mattn_go_sqlite3/sqlite3-binding.c -src external/com_github_mattn_go_sqlite3/sqlite3_opt_unlock_notify.c -src external/com_github_mattn_go_sqlite3/sqlite3-binding.h -src external/com_github_mattn_go_sqlite3/sqlite3ext.h -importpath github.com/mattn/go-sqlite3 -p github.com/mattn/go-sqlite3 -package_list bazel-out/host/bin/external/go_sdk_darwin/packages.txt -o bazel-out/darwin-opt/bin/external/com_github_mattn_go_sqlite3/go-sqlite3.a -x bazel-out/darwin-opt/bin/external/com_github_mattn_go_sqlite3/go-sqlite3.x -nogo bazel-out/darwin-opt-exec-2B5CBBC6/bin/vet_/vet -gcflags '' -asmflags '' -cppflags '-I external/com_github_mattn_go_sqlite3 -iquote .' -cflags '-D_FORTIFY_SOURCE=1 -fstack-protector -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O2 -DNDEBUG -DNS_BLOCK_ASSERTIONS=1 DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -mmacosx-version-min=11.1 -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" -target x86_64-apple-macosx -DHAVE_USLEEP=1 -DSQLITE_DEFAULT_WAL_SYNCHRONOUS=1 -DSQLITE_DISABLE_INTRINSIC -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS4_UNICODE61 -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_UPDATE_DELETE_LIMIT -DSQLITE_OMIT_DEPRECATED -DSQLITE_THREADSAFE=1 -DSQLITE_TRACE_SIZE_LIMIT=15 -Wno-deprecated-declarations -std=gnu99 -I. -fPIC' -cxxflags '-D_FORTIFY_SOURCE=1 -fstack-protector -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O2 -DNDEBUG -DNS_BLOCK_ASSERTIONS=1 -std=c++11 DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -mmacosx-version-min=11.1 -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" -target x86_64-apple-macosx -fPIC' -objcflags '-D_FORTIFY_SOURCE=1 -fstack-protector -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O2 -DNDEBUG -DNS_BLOCK_ASSERTIONS=1 DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -mmacosx-version-min=11.1 -D_FORTIFY_SOURCE=1 -fstack-protector -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O2 -DNDEBUG -DNS_BLOCK_ASSERTIONS=1 DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -mmacosx-version-min=11.1 -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" -target x86_64-apple-macosx -DHAVE_USLEEP=1 -DSQLITE_DEFAULT_WAL_SYNCHRONOUS=1 -DSQLITE_DISABLE_INTRINSIC -DSQLITE_ENABLE_FTS3 -DSQLITE_ENABLE_FTS3_PARENTHESIS -DSQLITE_ENABLE_FTS4_UNICODE61 -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_UPDATE_DELETE_LIMIT -DSQLITE_OMIT_DEPRECATED -DSQLITE_THREADSAFE=1 -DSQLITE_TRACE_SIZE_LIMIT=15 -Wno-deprecated-declarations -std=gnu99 -I. -fPIC' -objcxxflags '-D_FORTIFY_SOURCE=1 -fstack-protector -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O2 -DNDEBUG -DNS_BLOCK_ASSERTIONS=1 DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -mmacosx-version-min=11.1 -D_FORTIFY_SOURCE=1 -fstack-protector -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O2 -DNDEBUG -DNS_BLOCK_ASSERTIONS=1 -std=c++11 DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -mmacosx-version-min=11.1 -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" -target x86_64-apple-macosx -fPIC' -ldflags '-lc++ -fobjc-link-runtime -headerpad_max_install_names -no-canonical-prefixes -target x86_64-apple-macosx -mmacosx-version-min=11.1')
Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox
compilepkg: nogo: errors found by nogo during build-time code analysis:
/var/folders/7c/xbkf48h10g1f8qfvhjjlrjf00000gp/T/rules_go_work-066840393/_cgo_gotypes.go:1362:8: tautological condition: nil == nil (nilness)
Target //server/cmd/buildbuddy:buildbuddy failed to build
INFO: Elapsed time: 32.825s, Critical Path: 32.35s
INFO: 2 processes: 2 internal.
ERROR: Build failed. Not running target
INFO: Streaming build results to: https://app.buildbuddy.io/invocation/42be2f2c-0783-4f70-8bc1-eaba08c43848
FAILED: Build did NOT complete successfully

```

**Comment by siggisim** [comment]
Hey @lilidan - sorry about that. We recently added some static analyzers via nogo/govet that in this case is running on generated code. 

On linux this generated code ends up under the `/tmp` directory and is excluded, but that doesn't seem to be the case on Mac OS.

I've opened https://github.com/buildbuddy-io/buildbuddy/pull/436 that should fix this issue. I'll ping this thread once the fix is in.

Thanks for reporting!
